xonsh/tools.py

```diff
@@ -595,19 +584,16 @@ def ensure_int_or_slice(x):
 
 def is_string_set(x):
     """Tests if something is a set"""
-    if isinstance(x, string_types):
-        return False
-    else:
-        return (isinstance(x, Set) and
-                all([isinstance(a, string_types) for a in x]))
+    return (isinstance(x, Set) and
+            all(isinstance(a, str) for a in x))
 
 
 def csv_to_set(x):
     """Convert a comma-separated list of strings to a set of strings."""
     if not x:
         return set()
     else:
-        return set(x.split(','))
+        return set(y.strip() for y in x.split(','))
```

Comment:
Sure, but it would be incorrect to call it CSV then.  I think that there are only a couple of places in the code base that use this. If you want to change the names and the docs to reflect the strip, that is probably fine.